### ISSUE TITLE
bin/build results in "npm ERR! not ok code 0"

### ISSUE THREAD

**Issue by pdurbin** [metabase/metabase#1690]
Hi! [Metabase seems awesome](https://twitter.com/philipdurbin/status/672625807085469696) and I thought I'd take a quick swing at the steps in https://github.com/metabase/metabase/blob/master/docs/developers-guide.md but with a new checkout from master (commit 36b54ed) I'm getting `npm ERR! not ok code 0` from `bin/build`:

```
[pdurbin@tabby metabase]$ bin/build 
Tagging uberjar with version 'v0.13.1 36b54ed master 2015-12-15'...
Running 'npm install' to download javascript dependencies...
npm WARN engine get-stdin@5.0.0: wanted: {"node":">=0.12.0"} (current: {"node":"0.10.25","npm":"1.4.21"})
npm WARN optional dep failed, continuing fsevents@1.0.1
npm WARN optional dep failed, continuing fsevents@1.0.1
npm WARN optional dep failed, continuing fsevents@1.0.1
npm WARN optional dep failed, continuing fsevents@1.0.1
npm WARN optional dep failed, continuing fsevents@1.0.5
npm WARN optional dep failed, continuing fsevents@1.0.5
 
> caniuse-api@1.3.2 postinstall /home/pdurbin/github/metabase/metabase/node_modules/cssnext-loader/node_modules/cssnext/node_modules/caniuse-api
> node -e "require('shelljs/global');if(test('-d', 'dist'))exec('node dist/generate-features.js')"

npm WARN engine selenium-webdriver@2.47.0: wanted: {"node":">= 0.12.x"} (current: {"node":"0.10.25","npm":"1.4.21"})
npm WARN engine cryptiles@2.0.5: wanted: {"node":">=0.10.40"} (current: {"node":"0.10.25","npm":"1.4.21"})
npm WARN engine boom@2.9.0: wanted: {"node":">=0.10.40"} (current: {"node":"0.10.25","npm":"1.4.21"})
npm WARN engine hoek@2.16.3: wanted: {"node":">=0.10.40"} (current: {"node":"0.10.25","npm":"1.4.21"})
npm ERR! cb() never called!
npm ERR! not ok code 0
[pdurbin@tabby metabase]$ git log --oneline | head -1
36b54ed Merge pull request #1686 from metabase/fix-mysql-filter-not-equals-false
[pdurbin@tabby metabase]$ cat /etc/issue
Ubuntu 15.10 \n \l
```

What am I doing wrong? Is the version of node in Ubuntu 15.10 too old? Thanks in advance!

**Comment by agilliland** [comment]
your Ubuntu version shouldn't be a problem I wouldn't think.  looks like your node/npm versions are pretty old though.  i am currently running node 4.1.1 which uses npm 2.14.4

**Comment by tlrobinson** [comment]
This would likely be resolved by upgrading to npm v2. Try `npm install -g npm@2` then re-run the build.

**Comment by pdurbin** [comment]
I installed nodeenv and followed the instructions at https://pypi.python.org/pypi/nodeenv to create a new environment that has node 5.3.0 and npm 3.3.12. `bin/build` works just fine now. The jar was built, I ran `java -jar target/uberjar/metabase.jar` and was able to bring up the service at <http://localhost:3000>

I guess all the warnings I reported initially are enough to indicate that node and npm need to be newer than the packages that are provided even from an Ubuntu release that's only a couple months old. Thanks for listening, @agilliland and @tlrobinson . I appreciate it. I'll close this issue.
